Revise: some changes to Server GUI downloader handling (CID 1478854)

SlySven opened the pull request:
Initially this was going to address Coverity CID 1478854 - like Vadi's PR #5130. However I spotted some improvements that could be made as well:
* rename `(void) cTelnet::replyFinished(QNetworkReply*)` to `(void) cTelnet::slot_replyFinished(QNetworkReply*)` as it *is* a SLOT.
* rename `(QNetworkReply*) cTelnet::packageDownloadReply` to `(QNetworkReply*) mpPackageDownloadReply`
* check that the argument passed to the slot method matches the one that is expected (issues a warning if it doesn't)
* clean up the argument supplied to the slot method in ALL cases.
* reset the stored `QNetworkReply` point to a `nullptr` upon success (and thus returns it to the state it now has on class creation).

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: some changes to Server GUI downloader handling (CID 1478854)

Initially this was going to address Coverity CID 1478854 - like Vadi's
PR #5130. However I spotted some improvements that could be made as well:
* rename `(void) cTelnet::replyFinished(QNetworkReply*)` to
`(void) cTelnet::slot_replyFinished(QNetworkReply*)` as it *is* a SLOT.
* rename `(QNetworkReply*) cTelnet::packageDownloadReply` to
`(QNetworkReply*) mpPackageDownloadReply`
* check that the argument passed to the slot method matches the one that is
expected (issues a warning if it doesn't)
* clean up the argument supplied to the slot method in ALL cases.
* reset the stored `QNetworkReply` point to a `nullptr` upon success (and
thus returns it to the state it now has on class creation.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/ctelnet.cpp:
```diff
@@ -813,31 +813,38 @@ void cTelnet::sendTelnetOption(char type, char option)
-        return;
-    }
+    if (reply != mpPackageDownloadReply) {
+        qWarning().nospace().noquote() << "cTelnet::slot_replyFinished(QNetworkReply*) ERROR - received argument was not the expected stored pointer.";
```
I suggest adding more context for this - because the error will be vague otherwise:

```suggestion
        qWarning().nospace().noquote() << "cTelnet::slot_replyFinished(QNetworkReply*) ERROR - download finished, but it wasn't the one we are expecting";
```

vadi2 reviewed (approved):
Tested, installation works OK still :+1:

SlySven committed:
Update src/ctelnet.cpp

Revised warning message to give better context.

Signed-off by: Stephen Lyons <slysven@virginmedia.com>

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>
